gdraw/gmenu.c

```diff
@@ -2017,6 +1969,41 @@ int GMenuBarCheckKey(GWindow top, GGadget *g, GEvent *event) {
     dlist_free_external(&hklist);
 
 //    TRACE("menubarcheckkey(e1)\n");
+    return false;
+}
+
+
+int GMenuBarCheckKey(GWindow top, GGadget *g, GEvent *event) {
+    int i;
+    GMenuBar *mb = (GMenuBar *) g;
+    unichar_t keysym = event->u.chr.keysym;
+
+    if ( g==NULL || keysym==0 ) return( false ); /* exit if no gadget or key */
+
+    if ( (menumask&ksm_cmdmacosx) && keysym>0x7f &&
+	    (event->u.chr.state&ksm_meta) &&
+	    !(event->u.chr.state&menumask&(ksm_control|ksm_cmdmacosx)) )
+	keysym = GGadgetUndoMacEnglishOptionCombinations(event);
```

Comment:
I don't know. I just moved the hotkey code into a separate function. The diff doesn't really show this too well.